src/systems/odometry_publisher/OdometryPublisher.cc

```diff
@@ -278,6 +278,10 @@ void OdometryPublisher::PreUpdate(const gz::sim::UpdateInfo &_info,
 {
   GZ_PROFILE("OdometryPublisher::PreUpdate");
 
+  if (!this->dataPtr->model.Valid(_ecm)) {
+    return;
```

Comment:
What do you think about warning the user once that the OdometryPublisher has stopped? (see https://github.com/gazebosim/gz-sim/blob/633ce72171e27e83bf2e0292c9998e036d5da3fc/src/gui/plugins/environment_loader/EnvironmentLoader.cc#L153-L159)

Reply:
Yep, I copied solution from JointPositionController.